mlrun/run.py

```diff
@@ -1256,716 +1268,161 @@ def wait_for_runs_completion(runs: list, sleep=3, timeout=0, silent=False):
         )
 
-        # Log the zip file:
-        ctx.log_artifact(**logging_kwargs, item=key, local_path=directory_zip_path)
-
-        # Delete the zip file:
-        os.remove(directory_zip_path)
-
-    @staticmethod
-    def log_file(
-        ctx: MLClientCtx,
-        obj: Union[str, Path],
-        key: str,
-        logging_kwargs: dict,
-    ):
-        """
-        Log a file to MLRun.
-
-        :param ctx:            The MLRun context to log with.
-        :param obj:            The path of the file to log.
-        :param key:            The key of the artifact.
-        :param logging_kwargs: Additional keyword arguments to pass to the `context.log_artifact` method.
-
-        :raises MLRunInvalidArgumentError: In case the given path is not of a file or do not exist.
-        """
-        # In case it is a `pathlib` path, parse to str:
-        obj = str(obj)
-
-        # Verify the path is of an existing directory:
-        if not os.path.isfile(obj):
-            raise mlrun.errors.MLRunInvalidArgumentError(
-                f"The given path is not a file: '{obj}'"
-            )
-        if not os.path.exists(obj):
-            raise mlrun.errors.MLRunInvalidArgumentError(
-                f"The given directory path do not exist: '{obj}'"
-            )
-
-        # Log the zip file:
-        ctx.log_artifact(**logging_kwargs, item=key, local_path=os.path.abspath(obj))
-
-    @staticmethod
-    def log_object(ctx: MLClientCtx, obj, key: str, logging_kwargs: dict):
-        """
-        Log an object as a pickle.
-
-        :param ctx:            The MLRun context to log with.
-        :param obj:            The object to log.
-        :param key:            The key of the artifact.
-        :param logging_kwargs: Additional keyword arguments to pass to the `context.log_artifact` method.
-        """
-        ctx.log_artifact(
-            **logging_kwargs,
-            item=key,
-            body=obj if isinstance(obj, (bytes, bytearray)) else cloudpickle.dumps(obj),
-            format="pkl",
+    # Look for a builtin type (rest of the builtin types like `int`, `str`, `float` should be treated as results, hence
+    # not given as an input to an MLRun function, but as a parameter):
+    builtin_types = {
+        tuple.__name__: tuple,
+        list.__name__: list,
+        set.__name__: set,
+        dict.__name__: dict,
+        bytearray.__name__: bytearray,
+    }
+    if type_hint in builtin_types:
+        return builtin_types[type_hint]
+
+    # If it's not a builtin, its should have a full module path:
+    if "." not in type_hint:
+        raise mlrun.errors.MLRunInvalidArgumentError(
+            f"MLRun tried to get the type hint '{type_hint}' but it can't as it is not a valid builtin Python type "
+            f"(one of {', '.join(list(builtin_types.keys()))}). Pay attention using only the type as string is not "
+            f"allowed as the handler's scope is different then MLRun's."
```

Comment:
the error message doesn't states what is stated in the comment above, seems like you should also mention that it should have a full module path

Reply:
changed.